packages/ws-worker/src/channels/worker-queue.ts

```diff
@@ -72,7 +74,10 @@ const connectToWorkerQueue = (
       didOpen = true;
       shouldReportConnectionError = true;
 
-      const channel = socket.channel('worker:queue') as Channel;
+      // Build join payload with capacity if provided
+      const joinPayload = capacity !== undefined ? { capacity } : {};
```

Comment:
When would a worker NOT have capacity?

Ah, I see this is the result of bad typing in the server options, which I ought to raise an issue for. But capacity is defaulted and not really optional.

For now, I would simply do `const joinPayload = { capacity }`